### DISCUSSION TITLE
Unexpected token ':' in expression or statement.

### DISCUSSION THREAD

**Discussion by marioshaya** [JanDeDobbeleer/oh-my-posh#2779] [Troubleshoot] [answered]
Fan of oh-my-posh, i face this problem when I start windows terminal :
Unexpected token ':' in expression or statement.
At C:\Users\Mario\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1:2 char:14
+     "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-m ...
+              ~
Unexpected token ':' in expression or statement.
At C:\Users\Mario\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1:5 char:20
+         "alignment": "left",
+                    ~
Unexpected token ':' in expression or statement.
At C:\Users\Mario\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1:8 char:25
+             "background": "p:yellow",
+                         ~
Unexpected token ':' in expression or statement.
At C:\Users\Mario\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1:12 char:29
+               "display_host": false
+                             ~
Unexpected token ':' in expression or statement.
At C:\Users\Mario\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1:20 char:25
+             "background": "p:orange",
+                         ~
Unexpected token ':' in expression or statement.
At C:\Users\Mario\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1:24 char:26
+               "home_icon": "~",
+                          ~
Unexpected token ':' in expression or statement.
At C:\Users\Mario\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1:32 char:25
+             "background": "p:green",
+                         ~
Unexpected token ':' in expression or statement.
At C:\Users\Mario\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1:38 char:14
+             ],
+              ~
Missing argument in parameter list.
At C:\Users\Mario\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1:44 char:14
+             ],
+              ~
Missing argument in parameter list.
At C:\Users\Mario\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1:47 char:34
+               "branch_max_length": 25,
+                                  ~
Unexpected token ':' in expression or statement.
Not all parse errors were reported.  Correct the reported errors and try again.
    + CategoryInfo          : ParserError: (:) [], ParseException
    + FullyQualifiedErrorId : UnexpectedToken

How to resolve this please? Thanks

**Comment by JanDeDobbeleer** [comment, marked as answer]
@Qamard what's in your `$PROFILE`?

**Reply by marioshaya** [reply]
{
    "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
    "blocks": [
      {
        "alignment": "left",
        "segments": [
          {
            "background": "p:yellow",
            "foreground": "p:black",
            "leading_diamond": "\ue0b6",
            "properties": {
              "display_host": false
            },
            "style": "diamond",
            "template": " {{ if .SSHSession }}\uf817 {{ end }}{{ .UserName }} ",
            "trailing_diamond": "\ue0b0",
            "type": "session"
          },
          {
            "background": "p:orange",
            "foreground": "p:white",
            "powerline_symbol": "\ue0b0",
            "properties": {
              "home_icon": "~",
              "style": "folder"
            },
            "style": "powerline",
            "template": " \uf74a {{ path .Path .Location }} ",
            "type": "path"
          },
          {
            "background": "p:green",
            "background_templates": [
              "{{ if or (.Working.Changed) (.Staging.Changed) }}p:yellow{{ end }}",
              "{{ if and (gt .Ahead 0) (gt .Behind 0) }}p:red{{ end }}",
              "{{ if gt .Ahead 0 }}#49416D{{ end }}",
              "{{ if gt .Behind 0 }}#7A306C{{ end }}"
            ],
            "foreground": "p:black",
            "foreground_templates": [
              "{{ if or (.Working.Changed) (.Staging.Changed) }}p:black{{ end }}",
              "{{ if and (gt .Ahead 0) (gt .Behind 0) }}p:white{{ end }}",
              "{{ if gt .Ahead 0 }}p:white{{ end }}"
            ],
            "powerline_symbol": "\ue0b0",
            "properties": {
              "branch_max_length": 25,
              "fetch_status": true,
              "fetch_upstream_icon": true,
              "github_icon": "\uf7a3"
            },
            "style": "powerline",
            "template": " {{ if .UpstreamURL }}{{ url .UpstreamIcon .UpstreamURL }} {{ end }}{{ .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed }} \uf044 {{ .Working.String }}{{ end }}{{ if .Staging.Changed }} \uf046 {{ .Staging.String }}{{ end }} ",
            "type": "git"
          },
          {
            "background": "p:yellow",
            "foreground": "p:white",
            "powerline_symbol": "\ue0b0",
            "properties": {
              "root_icon": "\uf0ad"
            },
            "style": "powerline",
            "template": " \uf0e7 ",
            "type": "root"
          },
          {
            "background": "p:blue",
            "background_templates": [
              "{{ if gt .Code 0 }}p:red{{ end }}"
            ],
            "foreground": "p:white",
            "leading_diamond": "<transparent,background>\ue0b0</>",
            "properties": {
              "always_enabled": true
            },
            "style": "diamond",
            "template": " {{ if gt .Code 0 }}\uf00d{{ else }}\uf00c{{ end }} ",
            "trailing_diamond": "\ue0b4",
            "type": "exit"
          }
        ],
        "type": "prompt"
      },
      {
        "segments": [
          {
            "background": "transparent",
            "foreground": "p:green",
            "properties": {
              "display_mode": "files",
              "fetch_package_manager": false,
              "home_enabled": false
            },
            "style": "plain",
            "template": "\uf898 ",
            "type": "node"
          },
          {
            "background": "transparent",
            "foreground": "p:blue",
            "properties": {
              "fetch_version": false
            },
            "style": "powerline",
            "template": "\ufcd1 ",
            "type": "go"
          },
          {
            "background": "transparent",
            "foreground": "p:yellow",
            "properties": {
              "display_mode": "files",
              "fetch_version": true,
              "fetch_virtual_env": false
            },
            "style": "plain",
            "template": "\ue235 ",
            "type": "python"
          },
          {
            "background": "transparent",
            "foreground": "p:white",
            "style": "plain",
            "template": "in <p:blue><b>{{ .Name }}</b></> ",
            "type": "shell"
          },
          {
            "background": "transparent",
            "foreground": "p:white",
            "style": "plain",
            "template": "at <p:blue><b>{{ .CurrentDate | date \"15:04:05\" }}</b></>",
            "type": "time"
          }
        ],
        "type": "rprompt"
      }
    ],
    "console_title_template": "{{ .Shell }} in {{ .Folder }}",
    "final_space": true,
    "palette": {
      "black": "#262B44",
      "blue": "#4B95E9",
      "green": "#59C9A5",
      "orange": "#F07623",
      "red": "#D81E5B",
      "white": "#E0DEF4",
      "yellow": "#F3AE35"
    },
    "secondary_prompt": {
      "background": "transparent",
      "foreground": "p:black",
      "template": "<p:yellow,transparent>\ue0b6</><,p:yellow> > </><p:yellow,transparent>\ue0b0</> "
    },
    "tooltips": [
      {
        "background": "p:orange",
        "foreground": "p:white",
        "leading_diamond": "\ue0b6",
        "properties": {
          "display_default": true
        },
        "style": "diamond",
        "template": " \ue7ad {{ .Profile }}{{ if .Region }}@{{ .Region }}{{ end }} ",
        "tips": [
          "aws"
        ],
        "trailing_diamond": "\ue0b4",
        "type": "aws"
      },
      {
        "background": "p:blue",
        "foreground": "p:white",
        "leading_diamond": "\ue0b6",
        "style": "diamond",
        "template": " \ufd03 {{ .Name }} ",
        "tips": [
          "az"
        ],
        "trailing_diamond": "\ue0b4",
        "type": "az"
      }
    ],
    "transient_prompt": {
      "background": "transparent",
      "foreground": "p:black",
      "template": "<p:yellow,transparent>\ue0b6</><,p:yellow> {{ .Folder }} </><p:yellow,transparent>\ue0b0</> "
    },
    "version": 2
  }

**Reply by marioshaya** [reply]
I copied it from [https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json](url) cos I think it's only my setup that I did not touched was the source of the problem,

That's happened more than one month.

Thanks for replying

**Reply by JanDeDobbeleer** [reply]
@Qamard well, that makes sense. You added the config to your profile which definitely isn't the way to do things. I advice to take your time to read through the [guide](https://ohmyposh.dev/docs/installation/prompt), and remove everything from your profile for now to start fresh.

**Reply by marioshaya** [reply]
![eeeeeee](https://user-images.githubusercontent.com/66465021/189992064-1efb111c-2ada-4bda-b51c-ea4307894410.jpg)

**Reply by JanDeDobbeleer** [reply]
@Qamard please update the PSReadLine module or use PowerShell 7.

```powershell
Update-Module -Name PSReadLine  
```

**Reply by marioshaya** [reply]
It said 

Update-Module : Module 'PSReadLine' was not installed by using Install-Module, so it cannot be updated

I'm trying to install PowerShell 7 using 
winget install --id Microsoft.Powershell --source winget
winget install --id Microsoft.Powershell.Preview --source winget

**Reply by JanDeDobbeleer** [reply]
@Qamard please follow this [guide](https://github.com/PowerShell/PSReadLine#upgrading) to update.

**Reply by marioshaya** [reply]
Hello,
Installed pwsh 7.2.6
The problem is gone
But I can't save my profile
My setup is gone once I close windows terminal, and also it's not working in other tab

**Reply by JanDeDobbeleer** [reply]
You need to add the init line to your `$PROFILE` which differs between pwsh 7 and 5, so make sure to edit/create that file.

**Reply by marioshaya** [reply]
Thanks you very much, sir

Perfectly worked, thanks again.

**Reply by Mohamed-Abdulla** [reply]
> You need to add the init line to your `$PROFILE` which differs between pwsh 7 and 5, so make sure to edit/create that file.
how to add init line

**Reply by JanDeDobbeleer** [reply]
@Mohamed-Abdulla check [here](https://ohmyposh.dev/docs/installation/prompt) for detailed instructions.
